gammapy/modeling/scipy.py

```diff
@@ -173,4 +171,7 @@ def f(x):
     roots, res = find_roots(
         f, lower_bound=norm_best_fit, upper_bound=value_scan[-1], nbin=1, **kwargs
     )
-    return roots[0]
+    ul = roots[0]
+    if "converged" not in res[0].flag:
+        ul = np.nan
+    return ul
```

Comment:
same as above, roots[0] should be NaN if not converged

Reply:
should or are?

Reply:
They are initialized with Nan
https://github.com/gammapy/gammapy/blob/0643bb772c86092f758efad745f248a517658013/gammapy/utils/roots.py#L117
and replaced with value only if converged:
https://github.com/gammapy/gammapy/blob/0643bb772c86092f758efad745f248a517658013/gammapy/utils/roots.py#L132